core/src/main/java/io/aiven/klaw/service/ClusterApiService.java

```diff
@@ -754,28 +760,32 @@ public Map<String, String> retrieveMetrics(String jmxUrl, String objectName)
   private void setKwSSLContext() {
     if (keyStore != null && !keyStore.equals("null")) {
       TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
-      javax.net.ssl.SSLContext sslContext = null;
+      SSLContextBuilder builder = new SSLContextBuilder();
       try {
-        sslContext =
-            org.apache.http.ssl.SSLContexts.custom()
-                .loadKeyMaterial(getStore(keyStorePwd, keyStore), keyStorePwd.toCharArray())
-                .loadTrustMaterial(null, acceptingTrustStrategy)
+        builder
+            .loadKeyMaterial(getStore(keyStorePwd, keyStore), keyStorePwd.toCharArray())
+            .loadTrustMaterial(acceptingTrustStrategy);
+        SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(builder.build());
+        Registry<ConnectionSocketFactory> registry =
+            RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("http", new PlainConnectionSocketFactory())
+                .register("https", sslsf)
                 .build();
+        HttpClientConnectionManager poolingConnManager =
+            new PoolingHttpClientConnectionManager(registry);
+        CloseableHttpClient httpClient =
+            HttpClients.custom().setConnectionManager(poolingConnManager).build();
+        requestFactory = new HttpComponentsClientHttpRequestFactory();
+        requestFactory.setHttpClient(httpClient);
       } catch (NoSuchAlgorithmException
-          | KeyManagementException
           | KeyStoreException
           | CertificateException
           | UnrecoverableKeyException
           | IOException e) {
         log.error("Exception: ", e);
```

Comment:
Do we need to throw the other KeyStoreException  CertificateException  UnrecoverableKeyException IOException exceptions ?

Reply:
yea similar to cluster api implementation, if we dont throw, then we assume the connectivity could be fine, but later communication between apis would fail